ChatBot/ChatBot/Controller/ChatRoomViewController.swift

```diff
@@ -0,0 +1,208 @@
+                switch result {
+                case .success(let success):
+                    let answer = success.choices[0].message.content
+                    self.chats[self.chats.count - 1] = ChatBubble(message: Message(role: "assistant", content: answer))
+                    
+                    DispatchQueue.main.async {
+                        self.updateCollectionView()
+                        self.sendButton.isEnabled = true
+                    }
+                case .failure(let error):
+                    DispatchQueue.main.async {
+                        let alert = UIAlertController(title: "전송 실패", message: "통신 실패 네트워크 상태를 확인해주세요.", preferredStyle: .alert)
+                        
+                        alert.addAction(UIAlertAction(title: "확인", style: .default, handler: { _ in
+                            self.chats.removeLast()
+                            self.setUpSnapshot()
+                        }))
+                        self.present(alert, animated: true)
+                    }
+                    
+                }
+            }
+        }
+        
+        textInputView.text = nil
+        textInputView.insertText("")
+    }
+    
+    private func appendChat(content: String) {
+        chats.append(ChatBubble(message: Message(role: "user", content: content)))
+        chats.append(ChatBubble(message: Message(role: "assistant", content: "")))
+    }
+    
+    private func updateCollectionView() {
+        setUpSnapshot()
+        moveScroll()
+    }
+}
+
+// MARK: - UITextViewDelegate
+extension ChatRoomViewController: UITextViewDelegate {
+    
+    func textViewDidChange(_ textView: UITextView) {
+        sendButton.isEnabled = !textView.text.isEmpty
+        guard textView.contentSize.height <= view.frame.height * 0.1 else {
+            textView.isScrollEnabled = true
+            return
+        }
+        
+        textView.isScrollEnabled = false
+        
+        textView.constraints.forEach { (constraint) in
+            guard constraint.firstAttribute != .height else {
+                constraint.constant = textView.estimatedHeight
+                return
+            }
+        }
+    }
+}
+
+// MARK: - CollectionView
+extension ChatRoomViewController {
+    private func createLayout() -> UICollectionViewLayout {
+        var config = UICollectionLayoutListConfiguration(appearance: .plain)
+        config.showsSeparators = false
+        return UICollectionViewCompositionalLayout.list(using: config)
+    }
+    
+    private func configurationCell() {
+        let userChatBubbleRegistration = UICollectionView.CellRegistration<BubbleCell, ChatBubble> { cell, indexPath, itemIdentifier in
+            cell.setBubbleCell(message: itemIdentifier.message)
+        }
+        let assistantChatBubbleRegistration = UICollectionView.CellRegistration<BubbleCell, ChatBubble> { cell, indexPath, itemIdentifier in
+            cell.setBubbleCell(message: itemIdentifier.message)
+        }
```

Comment:
코드가 중복되는데 하나만 써도 될 것 같아요~!
```suggestion
        let chatBubbleRegistration = UICollectionView.CellRegistration<BubbleCell, ChatBubble> { cell, indexPath, itemIdentifier in
            cell.setBubbleCell(message: itemIdentifier.message)
        }
```

Reply:
[cb5eb4f](https://github.com/tasty-code/ios-chat-bot/pull/23/commits/cb5eb4f3624362c4d6b741d9f6742daddcf76378) 에서 수정했습니다~!